File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/dto/project/id/ProjectPermId.js
/**
 * Project perm id.
 * 
 * @author pkupczyk
 */
define([ "stjs", "dto/common/id/ObjectPermId", "dto/project/id/IProjectId" ], function(stjs, ObjectPermId, IProjectId) {
	/**
	 * @param permId
	 *            Project perm id, e.g. "201108050937246-1031".
	 */
	var ProjectPermId = function(permId) {
		ObjectPermId.call(this, permId);
	};
	stjs.extend(ProjectPermId, ObjectPermId, [ ObjectPermId, IProjectId ], function(constructor, prototype) {
		prototype['@type'] = 'dto.project.id.ProjectPermId';
		constructor.serialVersionUID = 1;
	}, {});
	return ProjectPermId;
})

File: openbis_api/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/project/id/ProjectPermId.java
/*
 * Copyright 2013 ETH Zuerich, CISD
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package ch.ethz.sis.openbis.generic.asapi.v3.dto.project.id;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.id.ObjectPermId;
import ch.systemsx.cisd.base.annotation.JsonObject;

/**
 * Project perm id.
 * 
 * @author pkupczyk
 */
@JsonObject("dto.project.id.ProjectPermId")
public class ProjectPermId extends ObjectPermId implements IProjectId
{

    private static final long serialVersionUID = 1L;

    /**
     * @param permId Project perm id, e.g. "201108050937246-1031".
     */
    public ProjectPermId(String permId)
    {
        super(permId != null ? permId.toUpperCase() : null);
    }

    //
    // JSON-RPC
    //

    @SuppressWarnings("unused")
    private ProjectPermId()
    {
        super();
    }

}
